### PR TITLE
Added cie and pcie pagerduty services.

### PR DESCRIPTION
**Pull request by brianseek**
## Description
This PR adds the pager duty component to the wrapper component. Also in this PR are some updates to move the footer component into the wrapper, so that the footer can still be displayed when a downtime window is active. 

Lots of unit tests were refactored as a result of adding and moving things at the wrapper level.

**To test:**

- Set a maintenance window for either Staging: Day of Check-in or Staging: Pre-Check-In here: https://dsva.pagerduty.com/service-directory?direction=asc&query=&sort_by=name&team_ids=PPW0SBU
- Add an appointment via the scheduling tool
- Visit the appointment in the review environment deployed by this PR.
- Verify that the downtime message is displayed (*Note it can take up to 6 minutes to show)

## Original issue(s)
department-of-veterans-affairs/va.gov-team#46556
department-of-veterans-affairs/va.gov-team#46559

## Testing done
Added unit test to trigger downtime and updated many other tests.

## Screenshots
0bf87dd3826d11fc22de6465fc51e8ca review vetsgov-internal_health-care_appointment-pre-check-in_error_error=bad-token

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/hooks/useFormRouting.jsx [hunk 1]
```diff
@@ -44,8 +44,11 @@ const useFormRouting = (router = {}) => {
 
   const getCurrentPageFromRouter = useCallback(
     () => {
-      // substring to remove the leading /
-      return router.location.pathname.substring(1);
+      if (router?.location?.pathname) {
+        // substring to remove the leading /
+        return router.location.pathname.substring(1);
+      }
+      return null;
```
Comment (acrollet) [inline src/applications/check-in/hooks/useFormRouting.jsx:51]:
```suggestion
      // substring to remove the leading /
      return router?.location?.pathname.substring(1) || null;
```

either way 🙂

**Review by acrollet — COMMENTED** [review]
I gave this a first pass and it's looking great! I really like the simplification of moving Footer and BackToHome into the wrapper. I have a couple of pretty minor suggestions if you want to give them a look and see what you think about incorporating them before I move on to testing this.

File: src/applications/check-in/components/layout/Footer.jsx [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
   return (
     <footer>
       <h2
         data-testid="heading"
         className="vads-u-font-size--lg vads-u-padding-bottom--1 vads-u-border-bottom--3px vads-u-border-color--primary"
       >
-        {header ?? t('need-help')}
+        {t('need-help')}
       </h2>
       {app === APP_NAMES.PRE_CHECK_IN ? (
```
[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/check-in/components/layout/Footer.jsx:28]: sorry, this comment is actually about the `{app === APP_NAMES.PRE_CHECK_IN ? (` line, I swear that's what I clicked on in VS Code 🤷

[the remark this answers is not in the document]
Comment (brianseek) [reply src/applications/check-in/components/layout/Footer.jsx:28]: so pass that into the Wrapper to pass to Footer? That is an option, but then I think we would be selecting the app in more places than we are here. I don't think every component that uses Wrapper is currently selecting the app name.

File: src/applications/check-in/hooks/useFormRouting.jsx (hunk 1 above)
Comment (brianseek) [reply src/applications/check-in/hooks/useFormRouting.jsx:51]: LGTM

File: src/applications/check-in/components/layout/Wrapper.jsx [hunk 3]
```diff
@@ -22,6 +30,11 @@ const Wrapper = props => {
     ? 'vads-u-padding-y--2'
     : ' vads-u-padding-y--3';
 
+  const selectApp = useMemo(makeSelectApp, []);
+  const { app } = useSelector(selectApp);
+  const downtimeDependency =
+    app === 'dayOf' ? externalServices.cie : externalServices.pcie;
```
Comment (acrollet) [inline src/applications/check-in/components/layout/Wrapper.jsx:36]:
```suggestion
    app === APP_NAMES.CHECK_IN ? externalServices.cie : externalServices.pcie;
```

should it be this for consistency?

**Comment by acrollet** [comment]
Testing results:

Once I realized that I needed to set up downtime for the *staging* cie app in pagerduty, (🤦) the window showed correctly when attempting to check in:

Screen Shot 2022-09-21 at 08 44 46

However, it does not show on an error page:

Screen Shot 2022-09-21 at 08 44 28

Is this intentional?

File: src/applications/check-in/components/layout/Wrapper.jsx (hunk 3 above)
Comment (brianseek) [reply src/applications/check-in/components/layout/Wrapper.jsx:36]: oh yep, thanks for catching that!

**Comment by brianseek** [comment]
> Testing results:
> 
> Once I realized that I needed to set up downtime for the _staging_ cie app in pagerduty, (🤦) the window showed correctly when attempting to check in:
> 
> Screen Shot 2022-09-21 at 08 44 46
> 
> However, it does not show on an error page:
> 
> Screen Shot 2022-09-21 at 08 44 28
> 
> Is this intentional?

That is weird. I wonder if it is because our error boundary is around the downtime component. Also wonder if this matters?

**Comment by acrollet** [comment]
🤔 I kind of think it matters, if a user tries to check in and gets taken to an error page because of an API issue I think we still want to be able to show the downtime message so they know it's a system problem rather than specific to them.

**Comment by brianseek** [comment]
> 🤔 I kind of think it matters, if a user tries to check in and gets taken to an error page because of an API issue I think we still want to be able to show the downtime message so they know it's a system problem rather than specific to them.

Good point, I was thinking that this was only happening when you went to an error direct. But yeah if it gets routed to error, we should still see downtime. Ideally it wouldn't even route to the error at all since the downtime should short circuit that first

**Comment by acrollet** [comment]
> > 🤔 I kind of think it matters, if a user tries to check in and gets taken to an error page because of an API issue I think we still want to be able to show the downtime message so they know it's a system problem rather than specific to them.
> 
> Good point, I was thinking that this was only happening when you went to an error direct. But yeah if it gets routed to error, we should still see downtime. Ideally it wouldn't even route to the error at all since the downtime should short circuit that first

oh good point on not getting there but yeah it feels best to make sure it shows everywhere imo

**Comment by acrollet** [comment]
Working now for me on review env error message page:

Screen Shot 2022-09-23 at 10 21 17

and regular pages:

Screen Shot 2022-09-23 at 10 23 42

**Review by acrollet — APPROVED** [review]
lgtm! nice work!

**Review by theodur — APPROVED** [review]

**Pull request merged**